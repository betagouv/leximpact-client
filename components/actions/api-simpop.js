/*
  const endpoint = this.endpoint();
  const { reforme } = this.state;
  fetch(`${endpoint}/calculate/compare`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      deciles: true,
      reforme,
    }),
  })
    .then(response => response.json())
    .then((json) => {
      this.setState({ total_pop: json });
    });
*/
const fetchSimPop = () => () => {};

export default fetchSimPop;
