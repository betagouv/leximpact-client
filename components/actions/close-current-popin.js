import Router from "next/router";

const closeCurrentPopin = () => {
  Router.push("/");
};

export default closeCurrentPopin;
