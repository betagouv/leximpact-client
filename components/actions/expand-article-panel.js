const expandArticlePanel = name => ({
  type: "onExpandArticlePanel",
  name,
});

export default expandArticlePanel;
