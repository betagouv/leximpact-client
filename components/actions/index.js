/* eslint
  indent: [2, 2],
  semi: [2, "always"],
  react/jsx-indent: [2, 2,{indentLogicalExpressions: false}],
  react/jsx-indent-props: [2, 2],
  import/order: [2, {
    newlines-between: "always",
    groups: ["builtin", "external", "parent", "sibling", "index"]
  }]
*/
export * from "./routing";
