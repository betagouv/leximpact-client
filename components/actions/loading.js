export const loadingComplete = () => ({
    type: "onLoadingComplete",
})

export const loadingStart = () => ({
    type: "onLoadingStart",
})
