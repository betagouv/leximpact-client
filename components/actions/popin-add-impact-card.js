import Router from "next/router";

const showAddImpactCardPopin = () => {
  Router.push("/?popin=ajouter-carte-impact");
  return { type: null };
};

export default showAddImpactCardPopin;
