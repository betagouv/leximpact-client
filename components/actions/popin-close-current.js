import Router from "next/router";

const closeCurrentPopin = () => {
  Router.push("/");
  return { type: null };
};

export default closeCurrentPopin;
