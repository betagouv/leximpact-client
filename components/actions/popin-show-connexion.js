import Router from "next/router";

const showConnexionPopin = () => {
  Router.push("/?popin=connection");
  return { type: null };
};

export default showConnexionPopin;
