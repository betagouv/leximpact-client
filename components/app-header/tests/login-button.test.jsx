import { createShallow } from "@material-ui/core/test-utils";

import LoginButton from "../login-button";
import { showConnexionPopin } from "../../actions";

const MockButtonClickHandler = jest.fn();
jest.mock("next/router", () => ({ push: jest.fn() }));
jest.mock("../../actions", () => ({
  showConnexionPopin: jest.fn(),
}));

describe("components | app-header | index", () => {
  let shallow = null;
  beforeAll(() => {
    MockButtonClickHandler.mockClear();
    const options = { dive: true };
    shallow = createShallow(options);
  });

  describe("snapshot", () => {
    it("doit correspondre au snapshot existant", () => {
      const props = {};
      const wrapper = shallow(<LoginButton {...props} />);
      expect(wrapper).toBeDefined();
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("render", () => {
    it("au click sur le bouton on doit avoir appeler la methode showConnexionPopin", () => {
      const props = {};
      const wrapper = shallow(<LoginButton {...props} />);
      wrapper.simulate("click");
      expect(showConnexionPopin).toHaveBeenCalledTimes(1);
    });
  });
});
