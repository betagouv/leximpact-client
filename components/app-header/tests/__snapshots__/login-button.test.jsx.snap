// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | app-header | index snapshot doit correspondre au snapshot existant 1`] = `
<WithStyles(ForwardRef(Button))
  className="LoginButton-menuButtonRoot-2"
  onClick={[Function]}
>
  <VpnKeyIcon
    classes={
      Object {
        "root": "LoginButton-avatarIcon-1",
      }
    }
  />
  Se connecter
</WithStyles(ForwardRef(Button))>
`;
